packages/cli/src/util/create-git-meta.ts

```diff
@@ -68,31 +76,35 @@ function getLastCommit(directory: string): Promise<git.Commit> {
   return new Promise((resolve, reject) => {
     git.getLastCommit(
       (err, commit) => {
-        if (err) return reject(err);
+        if (err) {
+          return reject(normalizeError(err));
+        }
+
         resolve(commit);
       },
       { dst: directory }
     );
   });
 }
 
-export function isDirty(directory: string, output: Output): Promise<boolean> {
-  return new Promise(resolve => {
+export function isDirty(directory: string): Promise<boolean> {
+  return new Promise((resolve, reject) => {
     // note: we specify the `--no-optional-locks` git flag so that `git status`
     // does not perform any "optional" operations such as optimizing the index
     // in the background: https://git-scm.com/docs/git-status#_background_refresh
     exec(
       'git --no-optional-locks status -s',
       { cwd: directory },
       function (err, stdout, stderr) {
-        let debugMessage = `Failed to determine if Git repo has been modified:`;
-        stderr = stderr && stderr.trim();
-        if (err || stderr) {
-          if (err) debugMessage += `\n${err}`;
-          if (stderr) debugMessage += `\n${stderr.trim()}`;
-          output.debug(debugMessage);
-          return resolve(false);
+        if (err) {
+          return reject(err);
         }
+        if (stderr?.trim()) {
+          return reject(new Error(stderr));
+        }
```

Comment:
Nit: I think the `?.` lets the conditional work as you expect it too, but I much prefer explicit conditionals.
```suggestion
        if (stderr !== undefined) {
          return reject(new Error(stderr.trim()));
        }
```

Reply:
The `trim` in the version as written in the PR also allows whitespace-only values to not reject. We could make this a more explicit check, though, certainly.